kernel/src/process/process_vm/mod.rs

```diff
@@ -188,34 +166,11 @@ impl ProcessVm {
         envp: Vec<CString>,
         aux_vec: AuxVec,
     ) -> Result<()> {
-        let root_vmar: ProcessVmarGuard<'_> = self.lock_root_vmar();
-        self.init_stack
-            .map_and_write(root_vmar.unwrap(), argv, envp, aux_vec)
-    }
-
-    pub(super) fn heap(&self) -> &Heap {
-        &self.heap
-    }
-
-    #[cfg(target_arch = "riscv64")]
-    pub(super) fn vdso_base(&self) -> Vaddr {
-        self.vdso_base.load(Ordering::Relaxed)
-    }
-
-    #[cfg(target_arch = "riscv64")]
-    pub(super) fn set_vdso_base(&self, addr: Vaddr) {
-        self.vdso_base.store(addr, Ordering::Relaxed);
-    }
-
-    /// Clears existing mappings and then maps the heap VMO to the current VMAR.
-    pub fn clear_and_map(&self) {
-        let root_vmar = self.lock_root_vmar();
-        root_vmar.unwrap().clear().unwrap();
-        self.heap.alloc_and_map_vm(root_vmar.unwrap()).unwrap();
+        self.init_stack().map_and_write(self, argv, envp, aux_vec)
     }
 }
 
-impl ProcessVm {
+impl Vmar<Full> {
```

Comment:
And is it intended to keep these stuffs under `process_vm/mod.rs`? I'm okay with it in this PR since it can mininize the diff (otherwise the PR becomes hard to review), but I'd like to clarify the intention.

Reply:
These methods on `Vmar<Full>` should eventually be moved into the `vm/vmar` module, but I think it’s better to wait until we remove capabilities from Vmar. At the moment, methods in `vm/vmar` are implemented either on `Vmar<R: TRights>` (static_cap.rs) or on `Vmar<Rights>` (dyn_cap.rs). In this PR the methods are simply added to `Vmar<Full>`(just follow our previous implementation), so placing them in either static_cap.rs or dyn_cap.rs would not be appropriate.

Reply:
> wait until we remove capabilities from Vmar

Oh really, let's just do it. Filed https://github.com/asterinas/asterinas/pull/2527.